core/src/main/java/com/threerings/getdown/data/Application.java

```diff
@@ -961,12 +962,20 @@ public Process createProcess (boolean optimum)
         }
 
         // pass along our proxy settings
-        String proxyHost;
-        if ((proxyHost = System.getProperty("http.proxyHost")) != null) {
-            args.add("-Dhttp.proxyHost=" + proxyHost);
-            args.add("-Dhttp.proxyPort=" + System.getProperty("http.proxyPort"));
-            args.add("-Dhttps.proxyHost=" + proxyHost);
-            args.add("-Dhttps.proxyPort=" + System.getProperty("http.proxyPort"));
+        if (proxy != Proxy.NO_PROXY && proxy.type() == Proxy.Type.HTTP && proxy.address() instanceof InetSocketAddress) {
+            InetSocketAddress proxyAddr = (InetSocketAddress) proxy.address();
+            args.add("-Dhttp.proxyHost=" + proxyAddr.getHostString());
+            args.add("-Dhttp.proxyPort=" + proxyAddr.getPort());
+            args.add("-Dhttps.proxyHost=" + proxyAddr.getHostString());
+            args.add("-Dhttps.proxyPort=" + proxyAddr.getPort());
+        } else {
+            String proxyHost;
+            if ((proxyHost = System.getProperty("http.proxyHost")) != null) {
+                args.add("-Dhttp.proxyHost=" + proxyHost);
+                args.add("-Dhttp.proxyPort=" + System.getProperty("http.proxyPort"));
+                args.add("-Dhttps.proxyHost=" + proxyHost);
+                args.add("-Dhttps.proxyPort=" + System.getProperty("http.proxyPort"));
+            }
```

Comment:
you can rewrite it in a shorter way to avoid duplication:
```
String proxyHost = System.getProperty("http.proxyHost");
String proxyPort = System.getProperty("http.proxyPort");
if (proxyHost == null && proxy.type() == Proxy.Type.HTTP && proxy.address() instanceof InetSocketAddress) {
  InetSocketAddress proxyAddr = (InetSocketAddress) proxy.address();
  proxyHost = proxyAddr.getHostString();
  proxyPort = String.valueOf(proxyAddr.getPort());
}
if (proxyHost != null) {
  args.add("-Dhttp.proxyHost=" + proxyHost);
  args.add("-Dhttp.proxyPort=" + proxyPort);
  args.add("-Dhttps.proxyHost=" + proxyHost);
  args.add("-Dhttps.proxyPort=" + proxyPort);
}
```
there's no need to check for no_proxy, proxy.type is enough

Reply:
I think my version is better readable, but I have no problem to slim it down. Nevertheless, I would like to get feedback, if this PR has a chance to get into a release very soon. I want to avoid to build my own fork.